src/newrelic/processor.go

```diff
@@ -777,8 +874,11 @@ func NewProcessor(cfg ProcessorConfig) *Processor {
 		harvestErrorChannel:   make(chan HarvestError),
 		quitChan:              make(chan struct{}),
 		processorHarvestChan:  make(chan ProcessorHarvest),
-		appConnectBackoff:     limits.AppConnectAttemptBackoff,
-		cfg:                   cfg,
+		// We don't want data usage collection to ever block, so we create a
+		// sized buffer that will drop any excess data once filled
+		dataUsageChannel:  make(chan dataUsageInfo, 25),
```

Comment:
out of curiosity, why are you buffering 25 dataUsageInfo objects in the chanel?

Reply:
I just chose 25 for little reason. I wanted it to be big enough that it could miss a few harvests and still be fine, but small enough to not have any sizeable footprint. It would probably be good to put it into a settings file.

Reply:
Yeah, I don't think it's actively harmful, unless the memory footprint matters that much. How many potential writers are there to the channel? There is only 1 reader, right? Does the reader drain the channel on read, or just read one object?

Reply:
https://github.com/newrelic/newrelic-php-agent/pull/507/files#diff-d34a767f316aaeded794ac3a231f867d3ffb42de19365ed338eace17f5876d94R755 drains the channel.

There is only one reader, though it is technically possible for multiple if the thread is being starved and another harvest begins. A harvestAll directive creates 9-ish writers.

Reply:
Gotcha, makes sense to me. Thanks for answering my questions, it's so hard to cold read this kind of code. LGTM